src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -140,7 +140,9 @@ def updatePileup(self, doc, rseList=None, validate=True):
         :return: list of MSPileupError or empty list
         """
         rseList = rseList or []
-        pname = doc.get('pileupName', '')
+        pname = doc.get('customName', '')
```

Comment:
I think this is wrong, given that the document id is actually the `pileupName`.

Reply:
Alan, I think logic is right, i.e.
```
        pname = doc.get('customName', '')
        if pname == '':
            pname = doc.get('pileupName', '')
```
if non empty custom name should be used instead pileup one than it is correct logic unless you say explicitly that during update we should not rely on custom name. Please clarify which direction to go and if necessary I'll adjust the code accordingly.

Reply:
As of L146 below, which for clarity I paste here:
```
spec = {'pileupName': pname}
```
we are using the pileup name (or custom name) for data lookup in the database. That means, using the spec above but having `pname` as a custom pileup name will not yield any results from MongoDB.

Reply:
There is a typo in your code (cusomName). 

In addition, it might be useful to have a new log record like "Fetching pileup document with the following spec: {spec}",
such that we know which key and dataset is being used to retrieve data from MongoDB.